app/src/main/java/com/example/habitsmasher/PublicPrivateButtons.java

```diff
@@ -0,0 +1,131 @@
+package com.example.habitsmasher;
+
+import static android.content.ContentValues.TAG;
+
+import android.nfc.Tag;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+
+import com.google.android.material.tabs.TabLayout;
+
+public class PublicPrivateButtons {
+    /**
+     * This class handles the logic behind the public and private buttons, which are used to
+     * set a habit to either public or private.
+     *
+     * @author Cameron Matthew
+     * @version 1.0
+     */
+
+    private Button _publicButton;
+    private Button _privateButton;
+    private boolean _publicSelected;
+    private boolean _privateSelected;
+
+    /**
+     * This constructor is the default constructor, and is used in {@link com.example.habitsmasher.ui.dashboard.AddHabitDialog}.
+     * @param view The view that has the buttons.
+     */
+    public PublicPrivateButtons(View view){
+        // get the buttons
+        _publicButton = view.findViewById(R.id.public_button);
+        _privateButton = view.findViewById(R.id.private_button);
+
+        // set the public button as the default button.
+        _publicButton.performClick();
+        _publicButton.setClickable(false);
+        _publicSelected = true;
+        _privateSelected = false;
+        Log.d("BUTTONS", "Standard const., public="+_publicSelected+", private="+_privateSelected);
+        Log.d("BUTTONS", "Is button clickable? " + _publicButton.isClickable());
+
+    }
+
+    /**
+     * This constructor is used whenever a habit is being viewed after creation, such as in {@link com.example.habitsmasher.ui.dashboard.EditHabitFragment}.
+     * @param view The view that has the buttons.
+     * @param isPublic Whether the habit is public or not.
+     */
+    public PublicPrivateButtons(View view, boolean isPublic){
```

Comment:
Constructor for habits that have already been made, aka edit habit and view habit